@here/harp-mapview/lib/poi/PoiRenderer.ts

```diff
@@ -34,91 +31,86 @@ const neutralColor = new THREE.Color(1, 1, 1);
-     * @param renderOrder - RenderOrder of the batch geometry's [[Mesh]].
+     * Creates a `PoiBuffer`
+     * @param buffer -
+     * @param layer - The {@link TextCanvas} layer used to render the POIs.
      */
     constructor(
-        readonly mapView: MapView,
-        readonly scene: THREE.Scene,
-        readonly imageItem: ImageItem,
-        readonly renderOrder: number
+        readonly buffer: BoxBuffer,
+        readonly layer: TextCanvasLayer,
+        private readonly m_onDispose: () => void
     ) {}
 
     /**
-     * Initialize with the {@link @here/harp-datasource-protocol#ImageTexture}.
-     *
-     * @remarks
-     * Loads the image and sets up the icon size, the texture
-     * coordinates and material of the batch. Since image loading is done asynchronously, this
-     * batch cannot be rendered right away. MapView#update is being triggered if it loaded
-     * successfully.
+     * Increases this `PoiBuffer`'s reference count.
+     * @returns this `PoiBuffer`.
      */
-    init() {
-        if (this.boxBuffer === undefined) {
-            this.setup();
-        }
+    increaseRefCount(): PoiBuffer {
+        ++this.m_refCount;
+        return this;
     }
 
     /**
-     * Clean the `PoiRenderBufferBatch`, remove all icon boxes. Called before starting a new frame.
+     * Decreases this `PoiBuffer`'s reference count. All resources will be disposed when the
+     * reference count reaches 0.
+     * @returns this `PoiBuffer`.
      */
-    reset(): void {
-        if (this.boxBuffer === undefined) {
-            this.init();
-        }
-        this.boxBuffer!.reset();
-    }
+    decreaseRefCount(): PoiBuffer {
+        assert(this.m_refCount > 0);
 
-    /**
-     * Update the geometry with all the added boxes during the frame.
-     */
-    update(): void {
-        if (this.boxBuffer === undefined) {
-            this.init();
+        if (--this.m_refCount === 0) {
+            this.dispose();
         }
-        this.boxBuffer!.updateBufferGeometry();
+        return this;
     }
 
-    /**
-     * Update the info with the memory footprint caused by objects owned by the
-     * `PoiRenderBufferBatch`.
-     *
-     * @param info - The info object to increment with the values from this `PoiRenderBufferBatch`.
-     */
-    updateMemoryUsage(info: MemoryUsage) {
-        if (this.boxBuffer !== undefined) {
-            this.boxBuffer.updateMemoryUsage(info);
-        }
+    private dispose() {
+        this.layer.storage.scene.remove(this.buffer.mesh);
+        this.buffer.dispose();
+        this.m_onDispose();
     }
+}
+
+/**
+ * @internal
+ *
+ * The `PoiBatch` contains the geometry and the material for all POIs that share the same icon image
+ * ({@link @here/harp-datasource-protocol#ImageTexture}).
+ *
+ * There is a `PoiBatch` for every icon in a texture atlas, since the size of the icon in the atlas
+ * as well as the texture coordinates are specified in the `PoiBatch`.
+ */
+class PoiBatch {
```

Comment:
This is the old PoiRenderBufferBatch, but I've moved in all PoiBuffers with same image but different render order, which now share same material and texture (before we didn't share them for different render orders of the same image).